sycl/include/sycl/ext/oneapi/matrix/matrix-tensorcore.hpp

```diff
@@ -235,6 +263,28 @@ struct joint_matrix_load_impl<
                                  get_layout_id<Layout>());
       }
 
+    } else if constexpr (std::is_same<T, double>::value) {
+      if constexpr (Use ==
+                    sycl::ext::oneapi::experimental::matrix::matrix_use::a) {
+        __dmma_m8n8k4_ld_a(res.data, src.get(), stride,
+                           get_layout_id<Layout>());
+      } else if constexpr (Use == sycl::ext::oneapi::experimental::matrix::
+                                      matrix_use::b) {
+        __dmma_m8n8k4_ld_b(res.data, src.get(), stride,
+                           get_layout_id<Layout>());
+      } else if constexpr (Use == sycl::ext::oneapi::experimental::matrix::
+                                      matrix_use::accumulator) {
+        __dmma_m8n8k4_ld_c(res.data, src.get(), stride,
+                           get_layout_id<Layout>());
+      }
+    } else if constexpr (NumRows == 8 && NumCols == 128) {
+      int32_t *tileptr = reinterpret_cast<int32_t *>(src.get());
+      __bmma_m8n8k128_ld_a_b1(res.data, tileptr, stride,
+                              get_layout_id<Layout>());
+    } else if constexpr (NumRows == 128 && NumCols == 8) {
+      int32_t *tileptr = reinterpret_cast<int32_t *>(src.get());
+      __bmma_m8n8k128_ld_b_b1(res.data, tileptr, stride,
```

Comment:
what types are supported in bmad? only double and i32?

Reply:
NVPTX bmad requires that matrix elements are stored in 32 bit untyped registers, int32_t is used here because when the NVPTX builtins for these functions were created int32_t register arguments were defined (uint32_t can also be used but there is no difference as far as NVPTX backend is concerned.).  As far as the user is concerned, I think that they should work with uint32_t for bmad cases as in https://github.com/intel/llvm-test-suite/pull/760.

double is not supported as a register storage type for bmad in NVPTX and I did not create a case for the user to use double with bmad.